ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/EeaWeb3jNonceProvider.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.requesthandler.sendtransaction.transaction;
+
+import tech.pegasys.ethsigner.core.requesthandler.sendtransaction.NonceProvider;
+
+import java.io.IOException;
+import java.math.BigInteger;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.web3j.protocol.core.Request;
+import org.web3j.protocol.core.methods.response.EthGetTransactionCount;
+import org.web3j.protocol.eea.Eea;
+
+public class EeaWeb3jNonceProvider implements NonceProvider {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Eea eea;
+  private final String accountAddress;
+  private final String privacyGroupId;
+
+  EeaWeb3jNonceProvider(final Eea eea, final String accountAddress, final String privacyGroupId) {
```

Comment:
currently EthNonceProvider exists forever, whereas the EeaWEb3jNonceProvider only exists for the duration of a transaction .... do we want them to have the same lifetimes? (probably aim toward one per transaction)

Reply:
changed so that both are now created in the factory consistently